### PR TITLE
ICU-20924 Use pkg-config to generate the path to pkgdata.inc

### PR DESCRIPTION
**Pull request by hughmcmaster**
On systems without `icu-config` installed, `pkgdata` will fail because it uses the legacy script to generate the path to pkgdata.inc.

Example command: `pkgdata -p bin_mkltfs -m static -v foo.txt'

`pkg-config` is always available, so I've created a patch to use it instead of `icu-config`.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20924
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by srl295** [comment]
I like the concept, but:

> `pkg-config` is always available

Not quite true. Can you change this to try `pkg-config` first and fall back to `icu-config` ? thanks.

**Comment by hughmcmaster** [comment]
@srl295 Thanks for the review. I've updated the MR.

File: icu4c/source/tools/pkgdata/pkgdata.cpp [hunk 1]
```diff
@@ -2203,21 +2208,18 @@ static void loadLists(UPKGOptions *o, UErrorCode *status)
...
-    if(buf[strlen(buf)-1]=='\n')
-    {
-        buf[strlen(buf)-1]=0;
-    }
-
-    if(buf[0] == 0)
-    {
-        fprintf(stderr, "%s: icu-config: invalid response from icu-config (fix PATH or use -O option)\n", progname);
+    if (!buf[0]) {
+        fprintf(stderr, "%s: invalid response from pkg-config or icu-config\n", progname);
```
Comment (srl295) [inline icu4c/source/tools/pkgdata/pkgdata.cpp]: (Ah, i lost my code I was writing… ) 

Basically, I was suggesting to remember which of `pkgconfigCmd` or `icuconfigCmd` was last used .And then print out  something like this:

> "pkgdata: Could not locate config file. Could not parse the results from 'icu-config --incpkgdatafile'. Check paths or use the -O option to specify a pkgdata.inc file."

Comment (hughmcmaster) [reply icu4c/source/tools/pkgdata/pkgdata.cpp]: Sure. I've updated the PR with both changes.

File: icu4c/source/tools/pkgdata/pkgdata.cpp [hunk 2]
```diff
@@ -2158,41 +2158,46 @@ static void loadLists(UPKGOptions *o, UErrorCode *status)
...
+    return TRUE;
+}
+#endif
+
+/* Get path to pkgdata.inc. Try pkg-config first, falling back to icu-config. */
+static int32_t pkg_getPkgDataPath(UBool verbose, UOption *option) {
+#if U_HAVE_POPEN
+    static char buf[512] = "";
+    UBool pkgconfigIsValid = TRUE;
+    const char *pkgconfigCmd = "pkg-config --variable=pkglibdir icu-i18n";
```
[the remark this answers is not in the document]
Comment (hughmcmaster) [reply icu4c/source/tools/pkgdata/pkgdata.cpp]: Yes, `--variable` is fantastic. I'm now using `icu-uc`.

And we could, in another PR, look at adding those variables to the pkg-config file.

**Comment by hughmcmaster** [comment]
Hi @srl295, any chance you can review the latest version of this MR? Thank you.

**Pull request merged**